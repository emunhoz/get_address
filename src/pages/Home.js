import React from 'react'
import GlobalStle from '../styles/GlobalStyle'
import Header from '../components/header'
import GetAddress from '../components/forms/get-address'
import GoogleMap from '../components/google-map/GoogleMap'
import { getCep } from '../services/getCep'
import { getGeoCode } from '../services/getGeoCode'
import * as S from './styled'

const Home = () => {
  const [addressForm, setAddressForm] = React.useState({})
  const [geoCode, setGeoCode] = React.useState({})
  const [loading, setLoading] = React.useState(false)
  const [error, setError] = React.useState(false)

  const handleSubmit = async ({ cep }) => {
    try {
      setLoading(true)
      const respViaCep = await getCep(cep)
      const respGeoCode = await getGeoCode(cep)
      setAddressForm(respViaCep.data)
      setGeoCode(respGeoCode.data.results[0].geometry.location)
      setError(false)
      setLoading(false)
    } catch (err) {
      setError(true)
      setLoading(false)
    }
  }

  return (
    <>
      <GlobalStle />
      <Header>
        <GetAddress handleSubmitForm={handleSubmit} loading={loading} />
      </Header>

      <S.Main>
        <S.AddressList>
          <S.AddressStyle>{addressForm.logradouro}</S.AddressStyle>
          <p>{addressForm.bairro}</p>
          {addressForm.localidade && (
            <p>
              {addressForm.localidade} - {addressForm.uf}
            </p>
          )}
          <p>{addressForm.cep}</p>
        </S.AddressList>
        {error && (
          <S.AddressStyle data-test-id='message-not-found'>
            Nenhum endere√ßo encontrado!
          </S.AddressStyle>
        )}
        {!error && <GoogleMap {...geoCode} />}
      </S.Main>
    </>
  )
}

export default Home
