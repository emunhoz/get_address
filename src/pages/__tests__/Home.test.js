import React from 'react'
import { render } from '@testing-library/react'

import Home from '../Home'

describe('Component: Home page', () => {
  let wrapper

  beforeEach(() => {
    wrapper = render(<Home />)
  })

  it('render the component', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
