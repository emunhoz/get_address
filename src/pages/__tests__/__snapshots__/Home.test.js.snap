// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Home page render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-AykKC bVuHWI"
      >
        <form
          class="sc-AykKF wCUMQ"
        >
          <div
            class="sc-AykKG fEfnIn"
          >
            <input
              class="sc-AykKD iRUOHY"
              data-testid="input-search"
              id="cep"
              mask="11111-111"
              maxlength="9"
              name="cep"
              placeholder="Digite seu CEP"
              size="9"
              value=""
            />
          </div>
          <button
            class="sc-AykKE hBlOvK"
          >
            Buscar
          </button>
        </form>
      </header>
      <main
        class="sc-AykKJ Ijlwq"
      >
        <div
          class="sc-AykKK FCLqI"
        >
          <p
            class="sc-fzXfLO cDaYXS"
          />
          <p />
          <p />
        </div>
        <div>
          Loading...
        </div>
      </main>
    </div>
    <script
      src="https://maps.googleapis.com/maps/api/js?key=AIzaSyA6WLiraDtwQIUCYmhgnFjAam5dI-rPVMA&callback=loaderCB01580792112367&libraries=places&v=3&language=en"
      type="text/javascript"
    />
  </body>,
  "container": <div>
    <header
      class="sc-AykKC bVuHWI"
    >
      <form
        class="sc-AykKF wCUMQ"
      >
        <div
          class="sc-AykKG fEfnIn"
        >
          <input
            class="sc-AykKD iRUOHY"
            data-testid="input-search"
            id="cep"
            mask="11111-111"
            maxlength="9"
            name="cep"
            placeholder="Digite seu CEP"
            size="9"
            value=""
          />
        </div>
        <button
          class="sc-AykKE hBlOvK"
        >
          Buscar
        </button>
      </form>
    </header>
    <main
      class="sc-AykKJ Ijlwq"
    >
      <div
        class="sc-AykKK FCLqI"
      >
        <p
          class="sc-fzXfLO cDaYXS"
        />
        <p />
        <p />
      </div>
      <div>
        Loading...
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
